chain-connect/docs/Examples.md

```diff
@@ -0,0 +1,319 @@
+
+- `galachainAddress`: The GalaChain address of the wallet.
+- `privateKey`: The private key of the wallet.
+- `publicKey`: The public key of the wallet.
+- `ethAddress`: The Ethereum address of the wallet.
+- `mnemonic`: The mnemonic phrase used to generate the wallet (may be `null`).
+
+#### Creating and Registering a Random Wallet
+
+To create a random wallet and register it with a GalaChain service, use `createAndRegisterRandomWallet`:
+
+```typescript
+import { WalletUtils } from "@gala-chain/connect";
+
+// Replace with your registration endpoint, or use this to test stage
+const registrationEndpoint = "https://stage-galaswap.gala.com/v1/CreateHeadlessWallet";
+
+async function createAndRegisterWallet() {
+  try {
+    const randomWallet = await WalletUtils.createAndRegisterRandomWallet(registrationEndpoint);
+
+    console.log("Private Key:", randomWallet.privateKey);
+  } catch (error) {
+    console.error("Failed to create and register wallet:", error);
+  }
+}
+
+createAndRegisterWallet();
+```
+
+The `createAndRegisterRandomWallet()` function returns an object containing:
+
+- `galachainAddress`: The GalaChain address of the wallet.
+- `privateKey`: The private key of the wallet.
+- `publicKey`: The public key of the wallet.
+- `ethAddress`: The Ethereum address of the wallet.
+- `mnemonic`: The mnemonic phrase used to generate the wallet (may be `null`).
+
+**Example Usage in an Application:**
+
+```typescript
+import { WalletUtils } from "@gala-chain/connect";
+import { ref } from "vue";
+
+export const message = ref("");
+export const createdUser = ref<string | null>(null);
+
+export async function generateWallet() {
+  try {
+    const wallet = await WalletUtils.createAndRegisterRandomWallet(
+      "https://stage-galaswap.gala.com/v1/CreateHeadlessWallet"
+    );
+    message.value = `Wallet generated and registered. Address: ${wallet.ethAddress}`;
+    createdUser.value = wallet.ethAddress;
+
+    //This is unsafe, please don't use in production
+    console.log("Private Key:", wallet.privateKey);
+  } catch (error) {
+    message.value = "Failed to generate wallet!";
+    console.error(error);
+  }
+}
+```
+
+### Template Integration
+
+In your Vue component, you can add a button to generate a wallet:
+
+```html
+<template>
+  <div>
+    <button @click="generateWallet">Generate Wallet</button>
+  </div>
+</template>
+
+<script>
+  import { generateWallet } from "./your-code-from-above";
+
+  export default {
+    setup() {
+      return {
+        generateWallet,
+      };
+    }
+  };
+</script>
+```
+
+## Server-Side Signing
+
+GalaChain Connect supports server-side signing using a private key. This is useful for backend services that need to interact with GalaChain without user interaction.
+
+### Server-Side Example
+
+```typescript
+import { PublicKeyApi, ServerSigningClient } from "@gala-chain/connect";
+
+const privateKey = "your-private-key-here";
+const connection = new ServerSigningClient(privateKey);
```

Comment:
I feel like we should change this class name to a more generic "SigningClient". This can be used both in browser and server sides, and using it in the browser client is a great example of creating a temporary wallet in games or dapps.

Reply:
Swapped the name out!